src/component/common/typography/index.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { css } from "@emotion/react";
+import { DESIGN_SYSTEM_TEXT } from "@/style/variable";
+import { DESIGN_SYSTEM_COLOR } from "@/style/variable";
+
+type TypographyProps = {
+  as?: React.ElementType;
+  variant?: keyof typeof DESIGN_SYSTEM_TEXT;
+  color?: keyof typeof DESIGN_SYSTEM_COLOR;
+  children: string;
+};
```

Comment:
```suggestion
type TextTags = "span" | "p" | "h1" | "h2" | "h3" | "h4" | "h5" | "h6";

type TypographyProps = {
  as?: Extract<React.ElementType,TextTags>;
  variant?: keyof typeof DESIGN_SYSTEM_TEXT;
  color?: keyof typeof DESIGN_SYSTEM_COLOR;
  children: string;
};
```

현재는 모든 태그를 넣을 수 있도록 되어 있는데, 다음과 같이 TextTag로 타입을 한정 시키는 것도 괜찮아 보입니다~!!

Reply:
Extract라는 교집합 문법이 있군요!! 하나 또 알아갑니다~!!